lib/mutant/mutator/node/define.rb

```diff
@@ -13,11 +13,39 @@ class Define < self
         # @api private
         def dispatch
           emit_arguments_mutations
+          emit_optarg_mutations
           emit_body(N_RAISE)
           emit_body(nil)
           emit_body_mutations if body
         end
 
+        # Emit mutation with optional arguments as assignments in method
+        #
+        # @return [undefined]
+        #
+        # @api private
+        def emit_optarg_mutations
+          optargs, args = arguments.children.partition do |child|
+            child.type.equal?(:optarg)
+          end
+
+          return if optargs.empty?
+
+          lvasgns = optargs.map do |optarg|
+            s(:lvasgn, *optarg.children)
+          end
+
+          body_children = [*lvasgns, body].compact
```

Comment:
So this could alternatively be written as

```ruby
body_children = optargs.map do |optarg|
  s(:lvasgn, *optarg.children)
end

body_children << body if body
```

but I thought that didn't read as nice. Does that clear things up?